packages/gatsby/src/schema/schema.js

```diff
@@ -338,52 +338,54 @@ const mergeTypes = ({
   createdFrom,
   parentSpan,
 }) => {
-  // Only allow user or plugin owning the type to extend already existing type.
+  // The merge is considered safe when a user or a plugin owning the type extend this type
+  // TODO: add proper conflicts detection and reporting (on the field level)
   const typeOwner = typeComposer.getExtension(`plugin`)
-  if (
+  const isSafeMerge =
     !plugin ||
     plugin.name === `default-site-plugin` ||
     plugin.name === typeOwner
-  ) {
-    if (type instanceof ObjectTypeComposer) {
-      mergeFields({ typeComposer, fields: type.getFields() })
-      type.getInterfaces().forEach(iface => typeComposer.addInterface(iface))
-    } else if (type instanceof InterfaceTypeComposer) {
-      mergeFields({ typeComposer, fields: type.getFields() })
-    } else if (type instanceof GraphQLObjectType) {
-      mergeFields({
-        typeComposer,
-        fields: defineFieldMapToConfig(type.getFields()),
-      })
-      type.getInterfaces().forEach(iface => typeComposer.addInterface(iface))
-    } else if (type instanceof GraphQLInterfaceType) {
-      mergeFields({
-        typeComposer,
-        fields: defineFieldMapToConfig(type.getFields()),
-      })
-    }
 
-    if (isNamedTypeComposer(type)) {
-      typeComposer.extendExtensions(type.getExtensions())
+  if (!isSafeMerge) {
+    if (typeOwner) {
```

Comment:
```suggestion
    if (typeOwner && typeOwner !== `internal-data-bridge`) {
```
We should treat `internal-data-bridge` as built-in.

Reply:
I think we should actually move type definitions away from the `internal-data-bridge` to built-in types. Keeping `createResolvers` there is fine though. I will prepare a follow-up PR to demonstrate what I mean

Reply:
On the other hand, you are probably right and we should keep it in the same place where we create the site node and treat it as built-in here.

Reply:
https://github.com/gatsbyjs/gatsby/pull/26866 as an alternative